paimon-format/src/main/java/org/apache/paimon/format/parquet/reader/NestedPrimitiveColumnReader.java

```diff
@@ -549,6 +563,53 @@ private WritableColumnVector fillColumnVector(int total, List valueList) {
         }
     }
 
+    private WritableColumnVector fillColumnVectorWithNone(int total) {
+        boolean[] isNull = new boolean[total];
+        Arrays.fill(isNull, true);
+        switch (dataType.getTypeRoot()) {
+            case CHAR:
+            case VARCHAR:
+            case BINARY:
+            case VARBINARY:
+                return new HeapBytesVector(total, isNull);
+            case BOOLEAN:
+                return new HeapBooleanVector(total, isNull);
+            case TINYINT:
+                return new HeapByteVector(total, isNull);
+            case SMALLINT:
+                return new HeapShortVector(total, isNull);
+            case INTEGER:
+            case DATE:
+            case TIME_WITHOUT_TIME_ZONE:
+                return new HeapIntVector(total, isNull);
+            case FLOAT:
+                return new HeapFloatVector(total, isNull);
+            case BIGINT:
+                return new HeapLongVector(total, isNull);
+            case DOUBLE:
+                return new HeapDoubleVector(total, isNull);
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                return new HeapTimestampVector(total, isNull);
+            case DECIMAL:
+                PrimitiveType.PrimitiveTypeName primitiveTypeName =
+                        descriptor.getPrimitiveType().getPrimitiveTypeName();
+                switch (primitiveTypeName) {
+                    case INT32:
+                        HeapIntVector phiv = new HeapIntVector(total, isNull);
+                        return new ParquetDecimalVector(phiv, total);
+                    case INT64:
+                        HeapLongVector phlv = new HeapLongVector(total, isNull);
+                        return new ParquetDecimalVector(phlv, total);
+                    default:
+                        HeapBytesVector phbv = new HeapBytesVector(total, isNull);
+                        return new ParquetDecimalVector(phbv, total);
+                }
+            default:
```

Comment:
I wonder if all the existing types are covered.

Reply:
yes except the INT32 and INT64, other primitiveType should deserialize as HeapBytesVector